### PR TITLE
Refactors skill modifiers to use dictionaries

### PR DESCRIPTION
**Pull request by ghost**
:cl: floyd
code: skills now use dictionaries for modifiers
/:cl:

Ninja suggested I use activity datums, which I might do once this is merged and we see the need for more diversity

### CODE REVIEW

File: code/game/objects/items/stacks/medical.dm [hunk 1]
```diff
@@ -58,8 +58,8 @@
 		user.visible_message("<span class='green'>[user] applies \the [src] on [C]'s [affecting.name].</span>", "<span class='green'>You apply \the [src] on [C]'s [affecting.name].</span>")
 		var/brute2heal = brute
 		var/burn2heal = burn
-		if(user?.mind?.get_skill_speed_modifier(/datum/skill/medical))
-			var/skillmods = user.mind.get_skill_speed_modifier(/datum/skill/medical)
+		if(user?.mind?.get_skill_modifier(/datum/skill/medical, SKILL_SPEED_MODIFIER))
+			var/skillmods = user.mind.get_skill_modifier(/datum/skill/medical, SKILL_SPEED_MODIFIER)
```
Comment (SpaceManiac) [inline code/game/objects/items/stacks/medical.dm]: seems like you could avoid calling this twice

**Review by SpaceManiac — CHANGES_REQUESTED** [review]

File: code/game/objects/items/stacks/medical.dm (hunk 1 above)
Comment (Qustinnus) [reply code/game/objects/items/stacks/medical.dm]: good point

Comment (SpaceManiac) [reply code/game/objects/items/stacks/medical.dm]: top tier web edit

also good to see you again

Comment (Qustinnus) [reply code/game/objects/items/stacks/medical.dm]: thank you! 👍

**Pull request merged**